packages/app-headless-cms/src/admin/views/contentEntries/ContentEntry/FullScreenContentEntry/FullScreenContentEntry.tsx

```diff
@@ -0,0 +1,101 @@
+            return <Original />;
+        }
+
+        const { loading } = useContentEntry();
+        const [isRevisionListOpen, openRevisionList] = useState<boolean>(false);
+
+        return (
+            <FullScreenContentEntryProvider
+                openRevisionList={openRevisionList}
+                isRevisionListOpen={isRevisionListOpen}
+            >
+                <FullScreenContentEntryContainer>
+                    <FullScreenContentEntryHeader>
+                        <FullScreenContentEntryHeaderContent style={{ width: "33%" }}>
+                            <FullScreenContentEntryHeaderLeft />
+                        </FullScreenContentEntryHeaderContent>
+                        <FullScreenContentEntryHeaderContent>
+                            {/*
+                                Empty div to relocate Entry Form Header via React Portal in full-screen mode.
+                                Ensures layout flexibility without disrupting React context and state.
+                            */}
+                            <div id={"cms-content-entry-header-right"} />
+                        </FullScreenContentEntryHeaderContent>
+                    </FullScreenContentEntryHeader>
+                    {loading && <CircularProgress style={{ zIndex: 10 }} />}
+                    <FullScreenContentEntryContent>
+                        <FullScreenContentEntryContentFormWrapper>
+                            <FullScreenContentEntryContentFormInner>
+                                <Original />
+                            </FullScreenContentEntryContentFormInner>
+                        </FullScreenContentEntryContentFormWrapper>
+                    </FullScreenContentEntryContent>
+                    <RevisionListDrawer />
+                </FullScreenContentEntryContainer>
+            </FullScreenContentEntryProvider>
+        );
+    };
+});
+
+const FullScreenContentEntryFormDecorator = ContentEntry.ContentEntryForm.createDecorator(
+    Original => {
+        return function ContentEntryForm(props) {
+            if (!featureFlags.allowCmsFullScreenEditor) {
+                return <Original {...props} />;
+            }
+
+            return <Original {...props} className={FullScreenContentEntryContentFormInnerCss} />;
+        };
+    }
+);
+
+const FullScreenContentEntryFormHeaderDecorator =
+    ContentEntry.ContentEntryForm.Header.createDecorator(Original => {
+        return function ContentEntryFormHeader() {
+            if (!featureFlags.allowCmsFullScreenEditor) {
+                return <Original />;
+            }
+
+            const headerRightElement = document.getElementById("cms-content-entry-header-right");
+
+            if (!headerRightElement) {
+                return <Original />;
+            }
+
+            return createPortal(<Original />, headerRightElement);
+        };
+    });
+
+export const FullScreenContentEntry = () => {
+    return (
```

Comment:
Add feature flag here, and remove it from all other components:

```
if (!featureFlags.allowCmsFullScreenEditor) {
    return null;
}
````

If the feature is disabled, you simply don't decorate anything.